plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -305,10 +306,7 @@ private Type toTrinoType(IndexMetadata.Field metaDataField, boolean isArray)
             }
         }
         else if (type instanceof DateTimeType) {
-            if (((DateTimeType) type).getFormats().isEmpty()) {
-                return TIMESTAMP_MILLIS;
-            }
-            // otherwise, skip -- we don't support custom formats, yet
+            return TIMESTAMP_TZ_MILLIS;
```

Comment:
This is error-prone. There are some assumptions that consumers of a value of type `TIMESTAMP_TZ_MILLIS` will make but those won't be satisfied for custom formats. I'd suggest to look at how the Kafka connector represents varying formats of timestamps. 

For example see https://github.com/trinodb/trino/blob/03d8e5abc686c5c4a9f96ca14db07e2aed880174/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/encoder/json/format/SecondsSinceEpochFormatter.java#L27 and it's usages.

Reply:
I am not determined about support TIMESTAMP_MILLIS. In Elasticsearch, dates are converted to UTC (if the time-zone is specified) and stored as a long number representing milliseconds-since-the-epoch.
[https://www.elastic.co/guide/en/elasticsearch/reference/current/date.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/date.html)
Could we understand Elasticsearch only support TIMESTAMP_WITH_ZONE not TIMESTAMP?